server/svix-server/src/v1/endpoints/health.rs

```diff
@@ -1,12 +1,88 @@
 // SPDX-FileCopyrightText: © 2022 Svix Authors
 // SPDX-License-Identifier: MIT
 
-use axum::{http::StatusCode, routing::get, Router};
+use std::time::Duration;
 
-async fn health() -> StatusCode {
+use axum::{http::StatusCode, routing::get, Extension, Json, Router};
+use sea_orm::{query::Statement, ConnectionTrait, DatabaseBackend, DatabaseConnection};
+use serde::{Deserialize, Serialize};
+
+use crate::{
+    core::cache::{kv_def, Cache, CacheBehavior, CacheKey, CacheValue},
+    queue::{QueueTask, TaskQueueProducer},
+};
+
+async fn heartbeat() -> StatusCode {
     StatusCode::NO_CONTENT
 }
 
+#[derive(Debug, Deserialize, Serialize)]
+pub enum HealthStatus {
```

Comment:
I think the problem with using an enum here is that I'm not sure this is actually what we want.

We probably want (in terms of the output json) something like:

```json
{
    "database": {
        "status": "ok",
        "information": "Can be an error string on error, and 'OK' on success."
    },
    "queue": ...,
    "cache": ...,
}
```
We can also potentially add "latency" to the HealthStatus later on, and stuff like that.

Reply:
Would you prefer a struct, or an internally tagged enum via Serde? Both would result in that format, except with the enum, you could skip the information field or anything else irrelevant to an Ok response.

Reply:
I think I'd do a struct because we just want more control and we'll probably share the fields anyway, as in we always want information (the error message or a success string), we always want latency (when we add it), etc.

Status can just be an enum though.